azure-pipelines.yml

```diff
@@ -3,135 +3,258 @@
+
+    steps:
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Build Artifacts
+      inputs:
+        artifactName: Microsoft.Spark.Binaries
+        downloadPath: $(Build.ArtifactStagingDirectory)
+    
+    - task: CopyFiles@2
+      displayName: Copy .NET artifacts
+      inputs:
+        sourceFolder: $(Build.ArtifactStagingDirectory)/Microsoft.Spark.Binaries/BuildArtifacts/src/csharp/Microsoft.Spark/bin/$(buildConfiguration)
+        contents: |
+          **/*.dll
+          **/*.pdb
+        targetFolder: $(Build.SourcesDirectory)/src/csharp/Microsoft.Spark/bin/$(buildConfiguration)
+
+    - task: CopyFiles@2
+      displayName: Copy scala artifacts
+      inputs:
+        sourceFolder: $(Build.ArtifactStagingDirectory)/Microsoft.Spark.Binaries/BuildArtifacts/src/scala
+        contents: '**/*.jar'
+        targetFolder: $(Build.SourcesDirectory)/src/scala
+
+    - task: NuGetToolInstaller@0
+      inputs:
+        versionSpec: '4.9.2'
+
+    - task: MicroBuildSigningPlugin@2
+      displayName: Install MicroBuild plugin
+      inputs:
+        signType: $(_SignType)
+        zipSources: false
+        feedSource: https://dnceng.pkgs.visualstudio.com/_packaging/MicroBuildToolset/nuget/v3/index.json
+      env:
+        TeamName: $(_TeamName)
+      condition: and(succeeded(), in(variables['_SignType'], 'real', 'test'), eq(variables['Agent.Os'], 'Windows_NT'))
+    
+    - task: MSBuild@1
+      displayName: 'Restore Sign Tools'
+      inputs:
+        solution: eng/Tools.proj
+        msbuildArguments: /t:Restore
+        msbuildVersion: 15.0
+  
+    - task: MSBuild@1
+      displayName: 'Sign .NET binaries'
+      inputs:
+        solution: eng/Sign.proj
+        msbuildArguments:  /t:SignBinaries
+                          /p:SignSparkBinaries=true
+                          /p:SignAssetsDir=$(Build.SourcesDirectory)\src\csharp\Microsoft.Spark\bin\$(buildConfiguration)\
+                          /p:SignType=$(_SignType)
+        msbuildVersion: 15.0
+
+    - task: MSBuild@1
+      displayName: 'Sign .jar binaries'
+      inputs:
+        solution: eng/Sign.proj
+        msbuildArguments: /t:SignBinaries
+                          /p:SignJarBinaries=true
+                          /p:SignAssetsDir=$(Build.SourcesDirectory)\src\scala\
+                          /p:SignType=$(_SignType)
+        msbuildVersion: 15.0
+    
+    - task: MSBuild@1
+      displayName: 'Sign worker binaries'
+      inputs:
+        solution: eng/Sign.proj
+        msbuildArguments: /t:SignBinaries
+                          /p:SignWorkerBinaries=true
+                          /p:SignAssetsDir=$(Build.ArtifactStagingDirectory)\Microsoft.Spark.Binaries\Microsoft.Spark.Worker\
+                          /p:SignType=$(_SignType)
+        msbuildVersion: 15.0
+
+    - task: NuGetCommand@2
+      inputs:
+        command: pack
+        packagesToPack: '$(Build.SourcesDirectory)\src\csharp\Microsoft.Spark.nuspec'
```

Comment:
Why aren't we using `dotnet pack` here?

Reply:
Ah, I see we weren't before. So maybe this is a question for @imback82 and @rapoth

---
In reply to: [278200318](https://github.com/dotnet/spark/pull/6#discussion_r278200318) [](ancestors = 278200318)

Reply:
`dotnet pack` should be also fine. What will be the benefit over `nuget` in this case? Just curious.

Reply:
It is cross plat, you don't need to bootstrap nuget.exe when you already have dotnet, you can package a csproj, so no need to create nuspec.

Reply:
I've opened https://github.com/dotnet/spark/issues/36 for this.